augur/augur_client.py

```diff
@@ -150,58 +158,41 @@ def event_signature_to_name_map(self):
       ([ensure_outcome_info(outcome)
         for outcome in data['outcomes'] if outcome]))
 
-  async def start_historical_event_listener(self, event_name_list,
-      current_block, on_event, increment_block=1000):
-    '''Polls for all Augur events starting at from_block. Once the latest block
-    has been reached, the loop will close.
-
-    Args:
-      event_name_list (list[string]): List containing names of Augur events.
-      current_block (int): Ethereum block to start listener from.
-      on_event (callback function): Function to run when an Augur event occurs.
-      increment_block (int): The number of blocks from the current_block to
-                             create a range of blocks to filter from. There are
-                             many blocks that will not contain an Augur event,
-                             so setting this number to a high value will search
-                             from a larger pool of blocks each iteration.
-
-    Raise:
-      IOError: If there is an issue connecting to the Augur node.
-    '''
-    self._require_is_open()
-    tasks = [self._start_new_block_listener(),
-             self._fetch_historical_events(event_name_list, current_block,
-                                           on_event, increment_block)]
-    # Terminate tasks once the historical listener has completed.
-    _, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-    for task in pending:
-      task.cancel()
-
-  async def start_event_listener(self, event_name_list, on_event):
-    '''Polls for new Augur events.
+  def filter_blocks(self, start_block, end_block, filter, handler,
+      increment=1000):
+    '''Retrieve all event logs from the start_block to the end_block. Once the
+    end block has been reached, filter_blocks will start a realtime listener to
+    run the handler everytime a new event occurs.
 
     Args:
-      event_name_list (list[string]): List containing names of Augur events.
-      on_event (callback function): Function to run when an Augur event occurs.
+      start_block (int): Starting block number.
+      end_block (int): Ending block number. Can also be the string 'latest'
+                       which represents the latest Ethereum block.
+      filter (dict): Dictionary containing filter parameters.
+      handler (callback): Callback function for when an event log is found in
+                          the current block.
+      increment (int): Specifies the range of blocks from the current_block to
```

Comment:
What does ```increment``` do?

Reply:
`increment` is the value that determines the amount to increase the `toBlock` value in the filter dict by on each iteration. I'll change the name of this variable to `block_increment_size` to better describe what the variable does.

Reply:
The name is fine, I'm wondering what happens if your ```increment``` is instead ```end_block - start_block + 1```?

Reply:
Filtering the blockchain with too large of a range resulted in the Web3py websocket timing out. I could try increasing the timeout/removing it and see how long that process is. My initial reasoning was that since it would timeout given the range of the entire blockchain, I could traverse it at a reasonable pace with a value like 1000 blocks/iteration.